core/state_transition.go

```diff
@@ -565,7 +590,7 @@ func (st *StateTransition) distributeTxFees(eHardfork bool) error {
 
 	caller := &vmcontext.SharedEVMRunner{EVM: st.evm}
 	governanceAddress, err := contracts.GetRegisteredAddress(caller, params.GovernanceRegistryId)
-	if err != nil {
+	if err != nil && !st.evm.ChainConfig().Faker {
```

Comment:
Why do we need to do a `Faker` check here? This is a pretty tricky place to do conditional logic that's primarily for testing.

Reply:
This is for `TestEIP1559Transition`, which runs a transaction with baseFee = 3. Without setting the Faker, the transaction will have the same result of baseFee == 0.
And I'm not sure how to fulfill that test without setting it.